packages/query-core/src/types.ts

```diff
@@ -32,7 +32,9 @@ export type QueryFunction<
   T = unknown,
   TQueryKey extends QueryKey = QueryKey,
   TPageParam = never,
-> = (context: QueryFunctionContext<TQueryKey, TPageParam>) => T | Promise<T>
+> =
+  | ((context: QueryFunctionContext<TQueryKey, TPageParam>) => T | Promise<T>)
+  | SkipToken
```

Comment:
I can either leave it in here and handle it the same way the "options.enabled" is handled, or I can move it into react-query

Reply:
This breaks packages/query-core/src/infiniteQueryObserver.ts
can you take a look @TkDodo <3